### PR TITLE
feat: verification module for drand quicknet

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Part of https://github.com/ChainSafe/forest/issues/3763

This PR(originally part of https://github.com/ChainSafe/forest/pull/3873) implements the beacon verification logic for drand quicknet. 
Please read the [G1 - G2 swap](https://drand.love/blog/2023/10/16/quicknet-is-live/#g1-g2-swap) section for the details.

Changes introduced in this pull request:

- Implement quicket beacon verification with unit tests
- Refactor mainnet beacon verification to support batching with unit tests and replace bls_signatues::verify_messages which is designed to verify bls messages in tipsets instead of beacons

## Reference issue to close (if applicable)

Closes

## Other information and links

https://drand.love/blog/2023/10/16/quicknet-is-live/

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/beacon/signatures/mod.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+use bls_signatures::Error;
+use blstrs::{G1Affine, G1Projective, G2Affine, G2Projective};
+use group::{prime::PrimeCurveAffine, Curve};
+use rayon::prelude::*;
+
+// re-exports
+pub use bls_signatures::{PublicKey as PublicKeyOnG1, Signature as SignatureOnG2};
+
+const CSUITE_G1: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";
+const CSUITE_G2: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
```
Comment (LesnyRumcajs) [inline src/beacon/signatures/mod.rs:16]: Where do these come from?

**Review by LesnyRumcajs — APPROVED** [review]
It's some kind of Elvish, I can't read it.

File: src/beacon/signatures/mod.rs (hunk 1 above)
Comment (hanabi1224) [reply src/beacon/signatures/mod.rs:16]: https://www.ietf.org/archive/id/draft-irtf-cfrg-bls-signature-05.html#name-basic

Comment (hanabi1224) [reply src/beacon/signatures/mod.rs:17]: Added to the comment

**Review by lemmih — APPROVED** [review]

**Pull request merged**